src/model.cpp

```diff
@@ -168,13 +169,12 @@ std::string Model::doSerialisation(Format format) const
                                 // Get parent components.
                                 Component* component1 = static_cast<Component*>(variable->getParent());
                                 Component* component2 = static_cast<Component*>(equivalentVariable->getParent());
-                                // Do not serialise a variable's parent component in a connection if that variable no
-                                // longer exists in that component. Allow serialisation of one componentless variable.
-                                if (component1) {
```

Comment:
This is not a case, it is not possible to get here if there is not at least one component.